test/y2storage/blk_device_test.rb

```diff
@@ -637,4 +637,65 @@
       end
     end
   end
+
+  # This is mostly a direct forward to libstorage-ng, but this test is here
+  # because Storage::BlkDevice#find_by_any_name does not follow the usual
+  # libstorage-ng convention for exceptions.
```

Comment:
What usual convention and why does it not apply here?

Reply:
Raising `Storage::DeviceHasWrongType` or `Storage::DeviceNotFound` when looking for something that is not there. Raising `Storage::Exception` is kind of a... well... exception. :-)

Reply:
```Storage::Exception``` is raised when the devicegraph is not the probed devicegraph.

Reply:
Ah, ok. That was not clear to me from the libstorage-ng doc. I will update this.

Reply:
@aschnell I have one question about that restriction of being only used in the probed devicegraph. And I guess @jreidinger opinion is useful here.

Take a look at this code currently present in yast2-bootloader:
```ruby
devices = Y2Storage::BlkDevice.all(staging)
# As wires devices have identical udev devices as its multipath device,
# we remove possible multipath wires from the list since we want to translate
# udev device to multipath device and not wire
devices.reject! { |d| d.is?(:disk) && d.descendants.any? { |i| i.is?(:multipath) } }
device = devices.find { |i| i.udev_full_all.include?(dev) }
return device.name if device
```
Most of it could be basically replaced by a call to `find_by_any_name` so it would work for all device names (right now it fail for virtio-, like it happened to yast2-update). BUT that code in yast2-bootloader is executed at the end of installation, after the libstorage commit.

Does still make sense in that moment to enforce the usage of probed? At the end of installation, the set of block devices in probed may not match the system (and on the other hand, `staging` should match).

A second and less important question, what about that issue with the udev names of multipath and their wires? Is handled by `find_by_any_name`?

Reply:
For the update case, we can implement a hack in which yast2-update (which runs before the commit) stores in the userdata of each BlkDevice the udev name that was used for matching it (if any). Then the bootloader could use that in addition to `udev_full_all`.

But that sounds too much like a workaround.

Reply:
find_by_any_name only works on the probed devicegraph since the devices may
not exist in any other devicegraph and even those existing in other
devicegraphs but not in probed cannot be found (due to the system lookup).

For the duplictate use of udev names with multipath: It is the responsibility
of udev to generate the correct link in /dev.

Reply:
_find_by_any_name only works on the probed devicegraph since the devices may not exist in any other devicegraph and even those existing in other devicegraphs but not in probed cannot be found (due to the system lookup)._

So that's basically what I said. The real requisite is that the blk devices in the devicegraph are the same than the block devices in the system. That's true for probed ONLY before the commit. After a successful commit is the other way around. The list of devices in probed may be different from the system but the list on staging so match.